### DISCUSSION TITLE
Is git CDN and Git LFS  works together better?

### DISCUSSION THREAD

**Discussion by maheshs007** [git-lfs/git-lfs#5856] [General]
Will Git CDN and Git LFS works together?
If we have Git CDN set up and it caches lfs objects to as per git CDN technical features.. do we really need to use LFS 
Can someone answer this

**Comment by chrisd8088** [comment]
Hey, thanks for your question.  I'm not familiar with the "Git CDN" project you mention, myself; perhaps it's this one?

https://gitlab.com/grouperenault/git_cdn

You might want to reach out to the team behind the Git CDN project for more assistance.

That said, I suspect the generic answer here is going to be that a CDN, with or without caching, isn't going to permanently store anything for you, either Git objects or Git LFS objects.

So if you need to use Git LFS because your objects aren't accepted by your Git remote system, or if they are just large files and you don't want them in your Git repository, that's when you would choose to use Git LFS.  Whether your proxy or CDN temporarily caches the Git LFS objects for some period of time them is sort of beside the point.

I will add one caution, though: the Git and Git LFS projects both [recommend](https://git-scm.com/docs/gitfaq#proxy) [against](https://github.com/git-lfs/git-lfs/blob/fc61febe9cc2d9ddc6ffe3e8d1ae546512632552/docs/man/git-lfs-faq.adoc#faq-proxy) the use of proxies like CDNs unlike they can be guaranteed to not modify the traffic in any way.
